src/lsp/cobol_preproc/src_overlay.ml

```diff
@@ -129,7 +140,13 @@ let join_limits: manager -> limit * limit -> srcloc = fun ctx (s, e) ->
       if e.pos_cnum  = e'.pos_cnum &&   (* compare only the fields that matter *)
          e.pos_fname = e'.pos_fname
       then (Links.replace ctx.cache s (loc, e); loc)
-      else try_cache_from ~loc @@ Links.find ctx.over_right_gap e'
+      else
+      if e.pos_cnum  < e'.pos_cnum &&   (* detect that there is no exact match *)
```

Comment:
For a "cleaner" temporary fix (that would make the bug just "less frequent"), please guard this with `e.pos_cnum > 0`.